plugins/inputs/socket_listener/socket_listener_test.go

```diff
@@ -55,13 +58,14 @@ func TestSocketListener_unix(t *testing.T) {
 }
 
 func TestSocketListener_unixgram(t *testing.T) {
-	defer os.Remove("/tmp/telegraf_test.sock")
+	os.Create("/tmp/telegraf_test.sock")
 	sl := newSocketListener()
 	sl.ServiceAddress = "unixgram:///tmp/telegraf_test.sock"
 
 	acc := &testutil.Accumulator{}
 	err := sl.Start(acc)
 	require.NoError(t, err)
+	defer sl.Stop()
```

Comment:
This was added so that the goroutines are shut down and the unix socket file removed. Otherwise they just get killed on when go exits (finishes the tests), and the file is left around.